sympy/sets/fancysets.py

```diff
@@ -766,6 +766,15 @@ def _sup(self):
     def _boundary(self):
         return self
 
+    def as_relational(self, x):
+        """Rewrite a Range in terms of equalities and logic operators. """
+        from sympy.functions.elementary.integers import floor
+        i = (x - (self.inf if self.inf.is_finite else self.sup))/self.step
+        return And(
+            Eq(i, floor(i)),
+            x >= self.inf if self.inf in self else x > self.inf,
+            x <= self.sup if self.sup in self else x < self.sup)
```

Comment:
what happens to `self.sup in self` if both _sup_ and _inf_ are symbolic? Can this expression be cast to a Python `bool` or does it raise an exception?

Reply:
If they are both symbolic (which they can't be at present) then a more complicated Piecewise would have to be returned which depends on whether a bound is infinite or not. Also, in general, the Piecewise would have to account for whether the step is positive or negative and which bound is inf/sup.